core/app/c/[communitySlug]/pubs/[pubId]/edit/page.tsx

```diff
@@ -0,0 +1,114 @@
+import type { Metadata } from "next";
+
+import Link from "next/link";
+import { notFound, redirect } from "next/navigation";
+
+import type { CommunitiesId, PubsId } from "db/public";
+import { Capabilities } from "db/src/public/Capabilities";
+import { MembershipType } from "db/src/public/MembershipType";
+import { Button } from "ui/button";
+
+import { ContentLayout } from "~/app/c/[communitySlug]/ContentLayout";
+import { PageTitleWithStatus } from "~/app/components/pubs/PubEditor/PageTitleWithStatus";
+import { PubEditor } from "~/app/components/pubs/PubEditor/PubEditor";
+import { getPageLoginData } from "~/lib/authentication/loginData";
+import { userCan } from "~/lib/authorization/capabilities";
+import { getCommunityBySlug } from "~/lib/db/queries";
+import { getPubsWithRelatedValuesAndChildren } from "~/lib/server";
+
+export async function generateMetadata({
+	params: { pubId, communitySlug },
+}: {
+	params: { pubId: string; communitySlug: string };
+}): Promise<Metadata> {
+	const community = await getCommunityBySlug(communitySlug);
+	if (!community) {
+		return { title: "Community Not Found" };
+	}
+
+	const pub = await getPubsWithRelatedValuesAndChildren({
+		pubId: pubId as PubsId,
+		communityId: community.id as CommunitiesId,
+	});
+
+	if (!pub) {
+		return { title: "Pub Not Found" };
+	}
+
+	// TODO: replace with proper title https://github.com/pubpub/platform/issues/736
+	const title = pub.values.find((v) => /title/.test(v.fieldSlug))?.value;
+
+	if (!title) {
+		return { title: `Edit Pub ${pub.id}` };
+	}
+
+	return { title: title as string };
+}
+
+export default async function Page({
+	params,
+	searchParams,
+}: {
+	params: { pubId: PubsId; communitySlug: string };
+	searchParams: Record<string, string>;
+}) {
+	const { pubId, communitySlug } = params;
+
+	const { user } = await getPageLoginData();
+
+	const canUpdatePub = await userCan(
+		Capabilities.updatePubValues,
+		{
+			type: MembershipType.pub,
+			pubId,
+		},
+		user.id
+	);
+
+	if (!pubId || !communitySlug) {
+		return null;
+	}
+
+	if (!canUpdatePub) {
+		redirect(`/c/${communitySlug}/unauthorized`);
+	}
+
+	const community = await getCommunityBySlug(communitySlug);
```

Comment:
We should really get rid of this and rename the correct one, but could you use `findCommunityBySlug` instead? `getCommunityBySlug` still uses prisma and isn't cached!